compiler/z/codegen/FPTreeEvaluator.cpp

```diff
@@ -1125,14 +1127,13 @@ OMR::Z::TreeEvaluator::floatRemHelper(TR::Node * node, TR::CodeGenerator * cg)
    outlinedSlowPath->swapInstructionListsWithCompilation();
    
    generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, labelNotExact);
+   callNode->setChild(0, node->getFirstChild());
+   callNode->setChild(1, node->getSecondChild());
    TR::Linkage *linkage = cg->createLinkage(TR_System);
-   node->setSymbolReference(cg->symRefTab()->findOrCreateRuntimeHelper(node->getOpCodeValue() == TR::fcall ? TR_S390jitMathHelperFREM : TR_S390jitMathHelperDREM, false, false, false));
-   TR::Register *helperReturnRegister = linkage->buildSystemLinkageDispatch(node);
-   // In addition to returning the result register, above dispatch call also set node with result register.
-   // Main line sets result in targetRegister so in OOL we copy helperReturnRegister to targetRegistert and unset node register.
+   TR::Register *helperReturnRegister = linkage->buildSystemLinkageDispatch(callNode);
    generateRRInstruction(cg, node->getDataType() == TR::Float ? TR::InstOpCode::LER : TR::InstOpCode::LDR, node,  targetRegister, helperReturnRegister);
-   node->unsetRegister();
-   cg->stopUsingRegister(helperReturnRegister);
+   callNode->setReferenceCount(1);
+   cg->decReferenceCount(callNode);
```

Comment:
Why do we need this? Is the intent to decrement the reference count of the call's children? If so we should do this explicitly so it is obvious to the reader.

Reply:
No. We do not recursively decrease reference count of the node here. Reason behind the setting ref count to one and decrease it is because, nativeDispatch sets the node with the return register, by artificially setting ref count to 1 and decreasing it I assume will make sure that helperReturnRegister is no longer live.

Reply:
Ok, I understand. @r30shah this is not so obvious though just looking at the code. Would you agree it may be a good idea to leave a short one sentence comment as to why we are doing this?